utils/token.ts

```diff
@@ -0,0 +1,39 @@
+import { AccessToken, Prisma, PrismaClient } from "@prisma/client";
+
+type TokenArgs = Prisma.AccessTokenCreateInput;
+
+const prisma = new PrismaClient();
+
+/**
+ * Creates a new token if there is not token with the "associatedEmail" field.
+ * More on prisma's upsert method here: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upserthttps://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert
+ *
+ * @param token {TokenArgs} - Token that has to be created/updated.
+ * */
+const upsertToken = async (data: TokenArgs): Promise<AccessToken | Error> => {
+  try {
+    const upsertData = await prisma.accessToken.upsert({
+      where: {
+        associatedEmail: data.associatedEmail,
+      },
+      update: data,
+      create: data,
+    });
+
+    return upsertData;
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      if (error.code === "P2002") {
```

Comment:
This wont occur if we remove the unique constraint in then schema, as we will support multiple calendars and token types.

Reply:
Why are we explicitly handling this error?

Reply:
@yesyash ?